### PR TITLE
Updated websocket.c to notify once per event cycle

### PR DESCRIPTION
**Pull request by uvjustin**
Libwebsockets sometimes generates multiple LWS_CALLBACK_CLIENT_WRITEABLE events per lws_callback_on_writable_all_protocol() call: https://libwebsockets.org/lws-api-doc-master/html/md_READMEs_README_8coding.html
In my case I was actually getting notify callbacks after every keepalive ping.
The change I made just uses an unsigned short to keep track of the notify event number and matches that against the local copy in each websocket session, not sending when they are still the same.

### CODE REVIEW

File: src/websocket.c [hunk 1]
```diff
@@ -326,6 +331,7 @@ websocket(void *arg)
 	  write_events = events;
 	  events = 0;
 	  lws_callback_on_writable_all_protocol(context, &protocols[WS_PROTOCOL_NOTIFY]);
+	  events_counter++;
```
Comment (chme) [inline src/websocket.c:334]: Increment the `write_events_counter` after the assignment of `write_events = events`. That way we start with 1 for the first event notification which makes it a bit easier in the callback (no decrement necessary).

Comment (uvjustin) [reply src/websocket.c]: Thanks for the review..made the requested changes

File: src/websocket.c [hunk 2]
```diff
@@ -263,16 +266,18 @@ callback_notify(struct lws *wsi, enum lws_callback_reasons reason, void *user, v
       case LWS_CALLBACK_ESTABLISHED:
 	// Initialize session data for new connections
 	memset(session_data, 0, sizeof(struct ws_session_data_notify));
+	session_data->counter = write_events_counter; // set it -1 so we can still send during this cycle
```
Comment (chme) [inline src/websocket.c:269]: Sorry, one thing left ... can you remove the comment. It is now misleading.

Comment (uvjustin) [reply src/websocket.c]: Sorry forgot to do that..done.

**Comment by chme** [comment]
Thanks!

**Pull request merged**